### ISSUE TITLE
JsonValue gets serialised as a regular structure into JSON

### ISSUE THREAD

**Issue by RReverser** [capnproto/capnproto#665]
I've noticed that if I try to import `JsonValue` into `.capnp` for the purpose of constructing dynamic value and use as part of another structure, it's getting serialised as if it would be regular union and not as the original JSON value.

For example, if I have something like

```capnp
using import "/capnp/compat/json.capnp".JsonValue;

struct MyStruct {
  field @0 :JsonValue;
}
```

and then do

```c++
auto obj = myStruct.initField().initObject(1);
auto entry = obj[0];
entry.setName("key");
entry.setValue("value");
```

then it gets serialised as

```json
{"field": {"object": [{ "name": "key", "value": {"string": "value"} }]}}
```

and not as one would expect:

```json
{"field": {"object": {"key": "value"}}}
```

It's possible to work around this by registering custom handler for `JsonValue` that checks `.which()` and forwards corresponding entry to the builder as-is, but I think it makes sense for such serialisation to be a built-in default behaviour.

**Comment by RReverser** [comment]
@kentonv I suppose this can be closed now?

**Comment by kentonv** [comment]
Indeed, fixed by #700
